wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/UserService.java

```diff
@@ -58,293 +59,187 @@
-                          "with predefined name and password",
-                  response = UserDescriptor.class)
-    @ApiResponses({@ApiResponse(code = 201, message = "Created"),
+    @GenerateLink(rel = LINK_REL_USER)
+    @ApiOperation(value = "Create a new user", response = UserDto.class)
+    @ApiResponses({@ApiResponse(code = 201, message = "User successfully created, response contains created entity"),
                    @ApiResponse(code = 400, message = "Missed required parameters, parameters are not valid"),
                    @ApiResponse(code = 401, message = "Missed token parameter"),
-                   @ApiResponse(code = 403, message = "Invalid or missing request parameters"),
-                   @ApiResponse(code = 409, message = "Invalid token"),
-                   @ApiResponse(code = 500, message = "Internal Server Error")})
-    public Response create(@ApiParam(value = "New user")
-                           UserDescriptor userDescriptor,
-                           @ApiParam(value = "Authentication token")
+                   @ApiResponse(code = 500, message = "Couldn't create user due to internal server error")})
+    public Response create(@ApiParam("New user")
+                           UserDto userDto,
+                           @ApiParam("Authentication token")
                            @QueryParam("token")
                            String token,
-                           @ApiParam(value = "User type")
+                           @ApiParam("User type")
                            @QueryParam("temporary")
                            @DefaultValue("false")
                            Boolean isTemporary,
                            @Context
-                           SecurityContext context) throws ForbiddenException,
-                                                           BadRequestException,
+                           SecurityContext context) throws BadRequestException,
                                                            UnauthorizedException,
                                                            ConflictException,
-                                                           ServerException,
-                                                           NotFoundException {
-        if (!context.isUserInRole("system/admin") && !userSelfCreationAllowed) {
-            throw new ForbiddenException("Currently only admins can create accounts. Please contact our Admin Team for further info.");
+                                                           ServerException {
+        final User createdUser;
+        if (token == null) {
+            checkUser(userDto);
+            createdUser = userManager.create(userDto, isTemporary);
+        } else {
+            if (!userSelfCreationAllowed) {
+                throw new BadRequestException("User self creation is not allowed");
+            }
+            createdUser = userManager.create(fromToken(token), isTemporary);
```

Comment:
about new parameter, if there was a class parameter it's quite easy to add new settings
userManager.create(UserCreateParameter)

About role of manager it's up to you. Either it is a dummy manager or smart manager.

so as it's not the role you want to give to the manager I think that it would be better to delegate to service then (not being the user manager)